src/workers/csv.worker.ts

```diff
@@ -0,0 +1,132 @@
+import { Worker, Job } from 'bullmq'
+import Papa from 'papaparse'
+import { CreateUser } from '../users/schema/create-user.schema'
+import { csvValidation } from '../users/utils/csvValidation'
+import IORedis from 'ioredis'
+import { createJob, updateJob } from '../processing/processing.service'
+import 'dotenv/config'
+import { JobStatus } from '../processing/schema/job'
+import { FastifyInstance } from 'fastify'
+
+export const connection = new IORedis({
+  maxRetriesPerRequest: null,
+})
+
+export const csvWorkerSetup = (fastify: FastifyInstance) => {
+  if (!process.env.DATABASE_URL) {
+    throw new Error('DATABASE_URL is not set')
+  }
+
+  return new Worker(
+    'csvQueue',
+    async (job: Job) => {
+      await createJob(fastify.db, {
+        uuid: job.id as string,
+        name: job.name,
+        status: JobStatus.pending,
+        data: job.data.csv,
+        user_uuid: job.data.user_uuid,
+      })
+
+      await updateJob(fastify.db, {
+        uuid: job.id as string,
+        name: job.name,
+        status: JobStatus.inProgress,
+        data: job.data.csv,
```

Comment:
I don't know if we need to store the csv in data field throughout . 
Also one improvement would be to store the csv in the new s3 u created - for backup.